### PR TITLE
Add the spatial-temporal cube logic to zion module

### PR DESCRIPTION
**Pull request by JavierJia**
@JeremyLi28 @xixZ 
It's far from perfect. But you two can start reviewing ASAP. 
I'll address the following tasks in today
- [x] update UI with msg structure changes. 
- [x] test the summary view

### CODE REVIEW

File: neo/app/views/index.scala.html [hunk 1]
```diff
@@ -7,21 +7,6 @@
 
   <time-series id="chart"></time-series>
 
-  <div id='sidebar' ng-init="click = -1" ng-class="{toggled: click === 1}" ng-click="click = click*-1">
```
Comment (JavierJia) [inline neo/app/views/index.scala.html]: recover this sidebar!

File: zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewActor.scala [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    case UpdateViewMsg => {
+      updateView() onSuccess {
+        case x =>
+          lastUpdateTime = new DateTime()
+          updateMetaRecord()
+      }
+    }
+  }
+
+  context.system.scheduler.schedule(30 minutes, 30 minutes, self, UpdateViewMsg)
```
Comment (JavierJia) [inline zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewActor.scala]: change `30` to a parameter

Comment (JavierJia) [inline zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewActor.scala]: done

File: neo/app/views/index.scala.html (hunk 1 above)
Comment (JavierJia) [inline neo/app/views/index.scala.html]: done

File: neo/app/actors/CacheActor.scala [hunk 3]
```diff
@@ -15,14 +17,14 @@ import scala.util.{Failure, Success}
   * There is one cache per keyword
   */
 class CacheActor(val viewsActor: ActorRef, val usGeoGnosis: USGeoGnosis)
-                (val dataSet: DataSet, val keyword: String)
+                (val dataSet: String, val keyword: Option[String])
```
Comment (xixZ) [inline neo/app/actors/CacheActor.scala]: Can this support NewYorkGnosis? Or I need to write a separate one?

File: zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewActor.scala [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      case Success(store) => {
+        key = store.viewKey
+        sourceName = store.sourceName
+        summaryLevel = store.summaryLevel
+        startTime = store.startTime
+        updateCycle = store.updateCycle
+        lastVisitTime = store.lastVisitTime
+        lastUpdateTime = store.lastUpdateTime
+        visitTimes = store.visitTimes
+        self ! DoneInitializing
```
Comment (xixZ) [inline zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewActor.scala]: Could you explain this line? "self ! DoneInitializing". What is it doing?

File: neo/app/actors/CacheActor.scala (hunk 3 above)
Comment (JavierJia) [inline neo/app/actors/CacheActor.scala]: Good question. I think I should make it general to make both gnosis work.

File: zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewActor.scala (hunk 4 above)
Comment (JavierJia) [inline zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewActor.scala]: `!` means `tell`. So it said tell `self` the `DoneIntitializing` msg.

**Comment by JavierJia** [comment]
I'll go ahead and merge this one before it grows too big. Will fire a issue to track of the rest of the TODOs.

**Pull request merged**